src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -1045,22 +1050,21 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger? errorLogger, Cancella
                             // write out the file if we have an output path
                             if (hasGeneratedOutputPath)
                             {
-                                var path = Path.Combine(Arguments.GeneratedFilesOutputDirectory!, tree.FilePath);
                                 if (Directory.Exists(Arguments.GeneratedFilesOutputDirectory))
                                 {
-                                    Directory.CreateDirectory(Path.GetDirectoryName(path)!);
+                                    Directory.CreateDirectory(Path.GetDirectoryName(tree.FilePath)!);
```

Comment:
>tree.FilePath [](http://example.com/codeflow?start=84&length=13)

Minor: Consider extracting a `path` local - in part, to minimize the changes here.